downgrade

```diff
@@ -39,6 +39,12 @@ previously_installed() {
   ' "$PACMAN_LOG"
 }
 
+currently_installed() {
+  LC_ALL=C.UTF8 "$PACMAN" -Qi "$1" 2> /dev/null \
+  | tr '\n' '\t' \
+  | sed 's#^Name\s*: \([^\t]*\)\t.*Version\s*: \([^\t]*\).*$#\1-\2#'
```

Comment:
I think this way is a bit clear:

```sh
pacman -Qi downgrade | awk -F " : " '
  /^Name / { name=$2 };
  /^Version / { version=$2 };
  END { printf("%s-%s\n", name, version) }
'
```

```console
p -Qi downgrade | awk -F " : " '/^Name / { name=$2 }; /^Version / { version=$2 }; END { printf("%s-%s\n", name, version) }'
```

Unfortunately, it outputs ` - ` (not the empty string) for non-installed packages. I'm hopeful there's a way to address that if I keep thinking.